src/components/Filters/CompanyReceivedFilter.jsx

```diff
@@ -0,0 +1,205 @@
+  const [ fromDate, setFromDate ] = useState( initialFromDate );
+  const [ throughDate, setThroughDate ] = useState( initialThroughDate );
+  const dispatch = useDispatch();
+  const errorMessageText = "'From' date must be less than 'through' date";
+
+  const fromRef = useRef();
+  const throughRef = useRef();
+
+  useEffect( () => {
+    // put it in YYYY-MM-DD format
+    // validate to make sure it's not invalid
+    const validFromDate = dateFrom ? formatDate( dateFrom ) : '';
+    setFromDate( validFromDate );
+  }, [ dateFrom ] );
+
+  useEffect( () => {
+    const validThroughDate = dateThrough ? formatDate( dateThrough ) : '';
+    setThroughDate( validThroughDate );
+  }, [ dateThrough ] );
+
+
+  const handleClear = period => {
+    if ( period === 'from' ) {
+      dispatch( changeDates( fieldName, '', throughDate ) )
+    }
+    if ( period === 'through' ) {
+      dispatch( changeDates( fieldName, fromDate, '' ) )
+    }
+  }
+
+  const handleKeyDownFromDate = event => {
+    if ( event.key === 'Enter' ) {
+      fromRef.current.blur();
+    }
+  }
+
+  const handleKeyDownThroughDate = event => {
+    if ( event.key === 'Enter' ) {
+      throughRef.current.blur();
+    }
+  }
+
+  const errors = useMemo( () => {
+    const errs = [];
+    if ( dayjs( fromDate ).isAfter( throughDate ) ) {
+      errs.push( errorMessageText );
+    }
+    if ( dayjs( fromDate ).isBefore( minDate ) ) {
+      errs.push( "'From' date must be after " +
+          dayjs( minDate ).format( DATE_VALIDATION_FORMAT ) )
+    }
+    if ( dayjs( throughDate ).isAfter( maxDate ) ) {
+      errs.push( "'Through' date must be before " +
+          dayjs( maxDate ).format( DATE_VALIDATION_FORMAT ) )
+    }
+
+    return errs;
+  }, [ fromDate, throughDate ] )
+
+  const handleDateChange = () => {
+    let _throughDate = throughDate;
+    let _fromDate = fromDate
+    if ( _fromDate && !dayjs( fromDate ).isValid() ) {
+      fromRef.current.value = '';
+      _fromDate = ''
+    }
+    if ( _throughDate && !dayjs( throughDate ).isValid() ) {
+      throughRef.current.value = '';
+      _throughDate = ''
+    }
+    const isDateDifferent = dateFrom !== _fromDate ||
+        dateThrough !== _throughDate;
+    if ( isDateDifferent ) {
+      dispatch( changeDates( fieldName, _fromDate, _throughDate ) )
+    }
+  };
+
+  const inputFromClassName = useMemo( () => {
+    const style = [ 'a-text-input' ]
+    if ( dayjs( fromDate ).isBefore( minDate ) ||
+        dayjs( fromDate ).isAfter( throughDate ) ) {
+      style.push( 'a-text-input__error' )
+    }
+    return style.join( ' ' )
+  }, [ fromDate, throughDate ] );
+
+  const inputThroughClassName = useMemo( () => {
+    const style = [ 'a-text-input' ]
+    if ( dayjs( throughDate ).isAfter( maxDate ) ||
+        dayjs( throughDate ).isBefore( fromDate ) ) {
+      style.push( 'a-text-input__error' )
+    }
+    return style.join( ' ' )
+  }, [ fromDate, throughDate ] );
+
+  return <CollapsibleFilter title={title}
```

Comment:
Maybe move `<CollapsibleFilter` to its own line so the wrapping here isn't so 
weird.